x-pack/plugins/files/common/types.ts

```diff
@@ -91,15 +96,25 @@ export interface File<Meta = unknown> {
    * User provided metadata
    */
   meta: Meta;
+
   alt: undefined | string;
 
+  /**
+   * MIME type of the file content, e.g.: image/png.
+   */
+  mime: undefined | string;
+
+  extension: undefined | string;
+
```

Comment:
So, extension will hold the sub-type of the mime type, e.g. "png" out of "image/png". What do you think about having the same for the type of the mime type, e.g. a field that would store "image"?

So that when we need to filter all files for images, we would not need to execute a full text search like `image*` on the `mime` text field, instead we would have some field, like  `mimeType` of type `keyword` and we could execute a more efficient search using that.

Basically what I am thinking, maybe we could have three fields:

```ts
{
  mime: 'type/subttype;param=value',
  mimeType: 'type',
  mimeSubtype: 'subtype'
}
```

Or not store the full mime type at all, just the two main components:

```ts
{
  mimeType: 'type',
  mimeSubtype: 'subtype'
}
```

Or as per [schema design doc](https://docs.google.com/document/d/1engj9h6qK_xS5wdLIpYsC1aP6M48sfjRB0Orsm6558U/edit#heading=h.1qa27sd9q9cl):


```ts
{
  mime: 'type',
  extension: 'subtype'
}
```